src/researchhub_document/permissions.py

```diff
@@ -1,29 +1,52 @@
-from utils.permissions import AuthorizationBasedPermission
-from utils.http import PATCH, POST, PUT, GET, DELETE
+from django.contrib.contenttypes.models import ContentType
+
+from hub.models import Hub
+from hypothesis.related_models.hypothesis import Hypothesis
+from paper.models import Paper
+from researchhub_access_group.constants import EDITOR
 from researchhub_document.related_models.constants.document_type import (
-    HYPOTHESIS,
     DISCUSSION,
+    HYPOTHESIS,
+    PAPER,
 )
-from hypothesis.related_models.hypothesis import Hypothesis
 from researchhub_document.models import (
     ResearchhubPost
 )
+from researchhub_document.related_models.researchhub_unified_document_model import ResearchhubUnifiedDocument
+from utils.permissions import AuthorizationBasedPermission
+
 
 class HasDocumentCensorPermission(AuthorizationBasedPermission):
     message = 'Need to be author or moderator to delete'
 
     def is_authorized(self, request, view, obj):
         if request.user.is_authenticated is False:
             return False
+        
+        document_type = obj.document_type if obj.__class__ is \
+            ResearchhubUnifiedDocument else Paper
```

Comment:
Use `isinstance`

Reply:
Maybe I'm missing something but I'm not a fan of defaulting to Paper if not unified doc instance. I would prefer if we do explicit checks to help mitigate bugs in the future.